pkg/networkmanager/ip-settings.jsx

```diff
@@ -72,14 +72,16 @@ export const IpSettingsDialog = ({ topic, connection, dev, settings }) => {
     const model = useContext(ModelContext);
 
     const params = settings[topic];
-    const [addresses, setAddresses] = useState(params.addresses ? params.addresses.map(addr => ({ address: addr[0], netmask: addr[1], gateway: addr[2] })) : []);
+    const [addresses, setAddresses] = useState(params.address_data);
+    const [defaultGateway, setDefaultGateway] = useState(params.gateway);
+    const [gatewaySetExplicitly, setGatewaySetExplicitly] = useState(false);
     const [dialogError, setDialogError] = useState(undefined);
-    const [dns, setDns] = useState(params.dns || []);
+    const [dns, setDns] = useState(params.dns_data || []);
```

Comment:
This added line is not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-22330-e10cfba6-20250915-113603-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)